pkg/virt-operator/webhooks/kubevirt-update-admitter.go

```diff
@@ -57,8 +58,10 @@ func (admitter *KubeVirtUpdateAdmitter) Admit(ar *v1beta1.AdmissionReview) *v1be
 		return resp
 	}
 
+	results = validateCustomizeComponents(newKV.Spec.CustomizeComponents)
+
```

Comment:
did you by chance check what happens if you post an initial KubeVirt CR with invalid json for customize components?

I'm not 100% sure, but since we will miss the invalid json on the CREATE, but catch it on the UPDATE, it's possible the KubeVirt CR's status can't be written to either. I'm just not sure how all this is wired up.  If we can't write to the status due to a JSON field in customizeComponents being invalid, then we'd never be able to report the issue on the kubevirt CR. 

if you haven't, can you make sure the kubevirt status can still be updated if the customizeComponents is invalid?

Reply:
I doubled checked and If the the kubevirt cr is created with invalid syntax the system will not be deplyed and the status of the kubevirt CR is still able to be updated. 
```
reason: DeploymentFailed
status: "False"
```

The KubeVirt spec is unable to be updated with 

```
{"component":"virt-operator","kind":"","level":"error","msg":"Could not update the KubeVirt resource.","name":"kubevirt","namespace":"kubevirt","pos":"kubevirt.go:659","reason":"admission webhook \"kubevirt-update-validator.kubevirt.io\" denied the request: patch \"{\\\"name\\\": \\\"not-valid}\" is not valid JSON","timestamp":"2021-03-12T14:56:39.470732Z","uid":"05a8bc49-92f7-4a2f-8ea6-4c68f5d9960f"}
```

but the status is.